plainly-plugin/src/ui/components/form/UploadForm.tsx

```diff
@@ -1,50 +1,79 @@
 import FormData from 'form-data';
-import { useState } from 'react';
+import { useCallback, useContext, useState } from 'react';
 import { makeProjectZipTmpDir, removeFolder } from '../../../node';
-import { postFormData } from '../../../node/request';
 import Button from '../common/Button';
 import Description from '../typography/Description';
 import Label from '../typography/Label';
 import PageHeading from '../typography/PageHeading';
 
 import fs from 'fs';
-import { useNotifications, useSettings } from '@src/ui/hooks';
-import { LoaderCircleIcon } from 'lucide-react';
+import { useNotifications } from '@src/ui/hooks';
+import {
+  useEditProject,
+  useGetProjectDetails,
+  useUploadProject,
+} from '@src/ui/hooks/api';
+import { useProjectData } from '../../hooks/useProjectData';
+import type { Project } from '../../types/project';
 import Alert from '../common/Alert';
+import { AuthContext } from '../settings/AuthProvider';
 
 export default function UploadForm() {
+  const { apiKey } = useContext(AuthContext);
+  const [projectData, setProjectData] = useProjectData();
+  const { isLoading, data } = useGetProjectDetails(projectData?.id, apiKey);
+  const { isPending: isUploading, mutateAsync: uploadProject } =
+    useUploadProject();
+  const { isPending: isEditing, mutateAsync: editProject } = useEditProject();
+  const { notifySuccess, notifyError } = useNotifications();
+
+  const projectExists = !!(projectData?.id && data);
+  const removedFromDatabase = !!(projectData?.id && !data);
+
+  const revisionHistoryCount = data?.revisionHistory?.length || 0;
+  const badRevision = projectData?.revisionCount !== revisionHistoryCount;
+
   const [inputs, setInputs] = useState<{
     projectName?: string;
     description?: string;
     tags?: string[];
   }>({});
-  const [loading, setLoading] = useState(false);
-  const { notifySuccess, notifyError } = useNotifications();
-  const { getSettingsApiKey, loading: settingsLoading } = useSettings();
-
-  const apiKey = getSettingsApiKey();
-
-  const isLoading = loading || settingsLoading;
-  const disabled = isLoading || !apiKey?.key;
+  const [uploadMode, setUploadMode] = useState<'new' | 'edit'>();
+
+  const uploadModes = [
+    {
+      value: 'new',
+      label: 'Upload new',
+      checked: uploadMode === 'new' || !projectExists,
+      disabled: false,
+      onChange: () => setUploadMode('new'),
+    },
+    {
+      value: 'edit',
+      label: 'Re-upload existing',
+      checked: uploadMode === 'edit' || (projectExists && uploadMode !== 'new'),
+      disabled: !projectExists,
+      onChange: () => setUploadMode('edit'),
+    },
+  ];
+
+  const loading = isLoading || isUploading || isEditing;
+  const disabled =
+    loading || (data && !(data?.analysis?.done || data?.analysis?.failed));
```

Comment:
* not good
* should be disabled only in edit mode

`const disabled = loading || (editing && analysisPending);`